### PR TITLE
Reloading resources from "Debug" menu

### LINKED ISSUE #2901: Add "refresh resources" button in dev mode
Restarting the game to test stuff like changing hitbox sizes, changing graphics, etc is really really annoying. A button in one of the dev mode menus to refresh all that stuff would be very helpful.

### PR DESCRIPTION
**Pull request by Vankata453**
The new "Reload Resources" option in the "Debug" menu allows for reloading all fonts, textures, sprites and tilesets.

**Additionally:**

* Removes the `name` field for sprites. I honestly didn't know it existed before, and don't know what the point of it is, considering it's not referenced anywhere in the code, other than in logs, where a filename can be used instead.

Closes #2901.

### CODE REVIEW

**Review by Frostwithasideofsalt — APPROVED** [review]
Thank the gato gods 🙏🏼

File: src/supertux/tile_set.cpp [hunk 1]
```diff
@@ -55,6 +56,18 @@ TileSet::TileSet() :
   m_tiles[0] = std::make_unique<Tile>();
 }
 
+void
+TileSet::reload()
+{
+  m_autotilesets.clear();
+  m_thunderstorm_tiles.clear();
+  m_tiles.resize(1);
```
Comment (tobbi) [inline src/supertux/tile_set.cpp:64]: Why `1` in particular?

File: src/video/sdl_surface_ptr.hpp [hunk 2]
```diff
@@ -81,6 +81,12 @@ class SDLSurfacePtr final
     SDL_FreeSurface(m_surface);
     m_surface = surface;
   }
+  void reset(SDLSurfacePtr& other)
```
Comment (tobbi) [inline src/video/sdl_surface_ptr.hpp:84]: Add a new line before this.

Comment (Vankata453) [reply src/video/sdl_surface_ptr.hpp:84]: To preserve the initial tile with an ID of 0. I can add a comment for this.

File: src/supertux/tile_set.cpp (hunk 1 above)
Comment (Vankata453) [reply src/supertux/tile_set.cpp:64]: To preserve the initial tile with an ID of 0. I can add a comment for this.

File: src/video/sdl_surface_ptr.hpp (hunk 2 above)
Comment (Vankata453) [reply src/video/sdl_surface_ptr.hpp:84]: Is it needed, since it's an overload of the same function?

Comment (tobbi) [reply src/video/sdl_surface_ptr.hpp:84]: I'd still think so. Those things irk me.

**Review by tobbi — APPROVED** [review]

**Comment by tylerandari13** [comment]
My train of thought to test this was to load the game, and remove an image. If it works then it should display a missing texture, because the images are no longer there, which resulted in this:
```
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:84 Image '/res/industrial/gfx-enemy/duker/duker-body.png' doesn't exist. Attempting to load ".deprecated" version.
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:439 Couldn't load texture '/res/industrial/gfx-enemy/duker/duker-body.png' (now using dummy texture): Couldn't open '/res/industrial/gfx-enemy/duker/duker-body.deprecated.png': not found
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:84 Image '/res/industrial/gfx-enemy/duker/duker-end.png' doesn't exist. Attempting to load ".deprecated" version.
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:439 Couldn't load texture '/res/industrial/gfx-enemy/duker/duker-end.png' (now using dummy texture): Couldn't open '/res/industrial/gfx-enemy/duker/duker-end.deprecated.png': not found
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:84 Image '/res/industrial/gfx-enemy/duker/duker-head.png' doesn't exist. Attempting to load ".deprecated" version.
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:439 Couldn't load texture '/res/industrial/gfx-enemy/duker/duker-head.png' (now using dummy texture): Couldn't open '/res/industrial/gfx-enemy/duker/duker-head.deprecated.png': not found
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:84 Image '/res/industrial/gfx-enemy/duker/duker-head.png' doesn't exist. Attempting to load ".deprecated" version.
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:376 Couldn't load texture '/res/industrial/gfx-enemy/duker/duker-head.png' (now using dummy texture): Couldn't open '/res/industrial/gfx-enemy/duker/duker-head.deprecated.png': not found
[FATAL] /home/tylerandari12/STcompilation/supertux/src/supertux/main.cpp:745 Unexpected exception: Cannot load image.
```
(The game closed after this was printed.)

`duker-body.png`, `duker-end.png`, and `duker-head.png` are the three images I removed so i could test. `duker-head.png` was the only image on a scripted object. Im not sure why the other two also tried to load.

I found another thing, which Im not sure if its related directly to this PR, but I might as well mention it. When this happened I thought "Well that sucks, but if I open the game again I should be able to add the images back in after opening and hit reload resources to make them appear.", but when I opened the level again, the scripted object which had the image was not there at all. I closed the game, added the images back in, and reopened the level, and the scripted object was back again.

**Comment by Vankata453** [comment]
> `duker-body.png`, `duker-end.png`, and `duker-head.png` are the three images I removed so i could test. `duker-head.png` was the only image on a scripted object. Im not sure why the other two also tried to load.

It reloads everything.

> I found another thing, which Im not sure if its related directly to this PR, but I might as well mention it. When this happened I thought "Well that sucks, but if I open the game again I should be able to add the images back in after opening and hit reload resources to make them appear.", but when I opened the level again, the scripted object which had the image was not there at all. I closed the game, added the images back in, and reopened the level, and the scripted object was back again.

Doesn't matter where the script object is, the resource is loaded into the game separately in its own manager.

File: src/supertux/menu/debug_menu.cpp [hunk 3]
```diff
@@ -71,6 +75,15 @@ DebugMenu::DebugMenu() :
...
              [](bool value){ g_debug.set_use_bitmap_fonts(value); });
+  add_entry(_("Reload Resources"),
+    []{
+      Resources::load();
+      TextureManager::current()->reload();
+
+      SpriteManager::current()->reload();
+      TileManager::current()->reload();
+    })
+    .set_help(_("Reloads all fonts, textures, sprites and tilesets."));
```
Comment (MatusGuy) [inline src/supertux/menu/debug_menu.cpp:86]: The fonts don't reload.

Comment (Vankata453) [reply src/supertux/menu/debug_menu.cpp:86]: `Resources::current()->reload()`

**Review by MatusGuy — APPROVED** [review]
Fonts don't reload, but because most modders don't care about changing the fonts, this should be good to go.

**Comment by Vankata453** [comment]
It's not, because it's supposed to do that too. Will look into it.

**Comment by Vankata453** [comment]
TTF fonts now properly reload.

**Review by MatusGuy — APPROVED** [review]

**Pull request merged**